enhancements/monitoring/alerting-consistency.md

```diff
@@ -0,0 +1,195 @@
+---
+title: alerting-consistency
+authors:
+  - "@michaelgugino"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-02-03
+last-updated: 2021-02-03
+status: implementable
+---
+
+# Alerting Consistency
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Alerts are a mess.  There's not really formal guidelines to follow when
+implementing alerts for a given component.  These are the guidelines.
+
+## Motivation
+
+Improve Signal to noise of alerts.  Align alert levels "Critical, Warning, Info"
+with operational expectations to allow easier triaging.  Ensure current and
+future alerts have documentation covering problem/investigation/solution steps.
+
+### Goals
+
+* Define clear criteria for designating an alert 'critical'
+* Define clear criteria for designating an alert 'warning'
+* Define minimum time thresholds before prior to triggering an alert
+* Establish clear documentation guidelines
+* Outline operational triage of firing alerts
+
+### Non-Goals
+
+* Define specific alert needs
+* Implement user-defined alerts
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an OpenShift developer, I need to ensure that my alerts are appropriately
+tuned to allow end-user operational efficiency.
+
+#### Story 2
+
+As an OpenShift administrator, I need to have confidence that alerts are
+actually informative and actionalble to maintain my clusters.
+
+### Implementation Details/Notes/Constraints [optional]
+
+This enhancement will require some developers and administrators to rethink
+their concept of alerts and alert levels.  Many people seem to have an
+intuition about how alerts should work that doesn't necessarily match how
+alerts actually work or how others believe alerts should work.
+
+Formalizing this guidance into a type of template or specification will allow
+developers and administrators to speak a common language and have a common
+understanding, even if the concepts seem initially unituitive.
+
+### Risks and Mitigations
+
+People will make wrong assumptions about alerts and how to handle them.  This
+is not unique to alerts, people often believe certain components should behave
+in a particular way in which they do not.  This is an artifact of poor
+documentation that leaves too much to the imagination.
+
+We can work around this problem with clear and concise documentation.
+
+## Design Details
+
+### Critical Alerts
+
+TL/DR:  Fix this now or cluster dies.
```

Comment:
Are we talking just about the cluster falling over, or would an alert about a situation that might lead to data loss without the cluster halting also qualify?

Reply:
I think the TLDR doesn't capture all the situations we might need a critical alert, I expanded upon some of these below.

Reply:
If the purpose of this doc is to provide guidance, I think it would help to phrase these sections more imperatively, as instructions to someone designing or implementing an alert. So here we might say

> Reserve critical level alerts only for reporting conditions that may lead to loss of data or inability to deliver service for the cluster as a whole. Failures of most individual components should not trigger critical level alerts, unless they would result in either of those conditions. Configure critical level alerts so they fire before the situation becomes irrecoverable. Expect users to be notified of a critical alert within a short period of time after it fires so they can respond with corrective action quickly.

and then below expand on that with the sorts of details that you have already with examples.